### ISSUE TITLE
Edit aiml error

### ISSUE THREAD

**Issue by MrAlicard** [Program-O/Program-O#346]
Hello
When I click to ID or I want search to something then I get these errors in editAJAX.error.log file and I don't get those things which I search to:
```
[28-Nov-2017 20:02:14 Europe/Budapest] PHP Warning:  Invalid argument supplied for foreach() in C:\xampp\htdocs\chatbot\admin\editAJAX.php on line 107
[28-Nov-2017 20:02:14 Europe/Budapest] PHP Warning:  Invalid argument supplied for foreach() in C:\xampp\htdocs\chatbot\admin\editAJAX.php on line 132
[28-Nov-2017 20:02:17 Europe/Budapest] PHP Warning:  Invalid argument supplied for foreach() in C:\xampp\htdocs\chatbot\admin\editAJAX.php on line 107
[28-Nov-2017 20:02:17 Europe/Budapest] PHP Warning:  Invalid argument supplied for foreach() in C:\xampp\htdocs\chatbot\admin\editAJAX.php on line 132

```

Before worked this with older version.

**Comment by Dave-Morton** [comment]
It seems that my last round of testing was incomplete, and I apologise for the inconvenience. I'll add this to the list of things to fix, and hope to get something to you shortly.

**Comment by MrAlicard** [comment]
I must reopen this because I can't edit the aiml files. If I give <srai> tag in template then I get this error top right side: There was an error updating the AIML - no changes made.
But if I go to logs then there is no logs about this problem.

**Comment by Dave-Morton** [comment]
Look in your browser's JavaScript console to see if there is something being reported there. Additionally, if you could provide the exact AIML category/categories that you cannot edit I'll add them to my chatbot and try to edit them, as I don't see the issue on my end.

**Comment by MrAlicard** [comment]
In console:
`Clicked!
editAiml.js:43:9
Cell class =  template
editAiml.js:47:9
current data = <srai>NEVEM <star index="1" /></srai> , old data = <pre>NEVEM</pre>
editAiml.js:31:9
current data = <srai>NEVEM <star index="1" /></srai>
editAiml.js:35:13
old data = <pre>NEVEM</pre>
editAiml.js:36:13
Saving the edit.
editAiml.js:102:5`


If you add in program-o these(in english):
That pattern: WELCOME * WHAT IS YOUR NAME
Pattern: My name is * but you can call me _
Template: <srai>NEVEM <star index="1" /></srai>

This problem appear with simple <srai>SOMETHING</srai> method also.

**Comment by Dave-Morton** [comment]
I'll check it out, thanks.

**Comment by Dave-Morton** [comment]
I see what you're referring to. It's taking out any "tags" it finds, for some reason. What started out as `<srai>NEVEM <star index="1"/></srai>` was changed into just `NEVEM`. I verified that this change is made within the DB as well, so something is actually removing those tags. I'll trace it out and see what's going on.

**Comment by Dave-Morton** [comment]
DRAT! I failed to reference this issue in my last push. I'm so sorry. I've found and corrected this issue last week, and pushed the fix for this bug (along with others). Please download the latest version from the `DEV` branch and give it a try. Let me know if this helps.

**Comment by MrAlicard** [comment]
Hello. Ok, I will test it.

**Comment by MrAlicard** [comment]
I downloaded the dev version and I tried put same srai, but I get same error.
If I click to Teach and give <srai>NEVEM <star index="1" /></srai> then in edit there is the srai, but if I try edit and put the srai then I get the sam error: `There was an error updating the AIML - no changes made.`

But I don't get log files.

**Comment by Dave-Morton** [comment]
Given that all of the "magic" in the Search/Edit panel (and others) is run by JavaScript, it's sort of necessary to make sure you clear your cache to get the new JS code to replace the cached version. If you've already cleared your browser cache before trying, then hopefully there will be something in either the logs folder, or the JS console that gives an indication as to why the fix isn't working for you.

**Comment by MrAlicard** [comment]
I didn't clear the cache before but I did now.
I tried with Firefox, Microsoft Edge and Chrome but same results.
I followed this tutorial also to clear cache in Firefox: https://support.mozilla.org/en-US/kb/how-clear-firefox-cache
I used to clear every cache in every browser and system with Advanced System Care and CCleaner but these didn't helped me.
In console:
`Clicked!
editAiml.js:43:9
Cell class =  template
editAiml.js:47:9
current data = <srai>NEVEM <star index="1" /></srai> , old data = <pre>test</pre>
editAiml.js:31:9
current data = <srai>NEVEM <star index="1" /></srai>
editAiml.js:35:13
old data = <pre>test</pre>
editAiml.js:36:13
Saving the edit.
editAiml.js:102:5`

**Comment by Dave-Morton** [comment]
Just returning the variable without filtering it is the same as using $_GET or $_POST directly, so it's a major security risk under certain circumstances. If you're using Program O on your local computer's web server _from_ your local computer then the risk is minimal. But if it's online somewhere, and others use the admin pages then the risk goes up considerably.

while looking at the code you posted, something in the back of my mind bothered me about it, so I looked at what I currently have in my local repo, and saw that that specific bit of code isn't the same as what you have there, so I did some more digging and found that while I __DID__ make a bug fix commit for this particular bug, it never got pushed, so I'm going to take care of that here in a few minutes. I first want to make sure that everything is good before I push it.
